modules/data_warehouse/main.tf

```diff
@@ -0,0 +1,396 @@
+  ]
+}
+
+# TODO: Add scheduled load job
+
+# Set up Storage Buckets
+# # Set up the raw storage bucket
+resource "google_storage_bucket" "raw_bucket" {
+  name                        = "ds-edw-raw-${random_id.id.hex}"
+  project                     = var.project_id
+  location                    = var.region
+  uniform_bucket_level_access = true
+  force_destroy               = true
+
+  # public_access_prevention = "enforced" # need to validate if this is a hard requirement
+}
+
+# # Set up the provisioning bucketstorage bucket
+resource "google_storage_bucket" "provisioning_bucket" {
+  name                        = "ds-edw-provisioner-${random_id.id.hex}"
+  project                     = var.project_id
+  location                    = var.region
+  uniform_bucket_level_access = true
+  force_destroy               = true
+
+  # public_access_prevention = "enforced"
+}
+
+# Set up BigQuery resources
+# # Create the BigQuery dataset
+resource "google_bigquery_dataset" "ds_edw" {
+  project       = var.project_id
+  dataset_id    = "ds_edw"
+  friendly_name = "My EDW Dataset"
+  description   = "My EDW Dataset with tables"
+  location      = var.region
+  labels        = var.labels
+}
+
+# # Create a BigQuery connection
+resource "google_bigquery_connection" "ds_connection" {
+  project       = var.project_id
+  connection_id = "ds_connection"
+  location      = var.region
+  friendly_name = "Storage Bucket Connection"
+  cloud_resource {}
+}
+
+# # Grant IAM access to the BigQuery Connection account for Cloud Storage
+resource "google_project_iam_member" "bq_connection_iam_object_viewer" {
+  project = var.project_id
+  role    = "roles/storage.objectViewer"
+  member  = "serviceAccount:${google_bigquery_connection.ds_connection.cloud_resource[0].service_account_id}"
+
+  depends_on = [
+    google_bigquery_connection.ds_connection
+  ]
+}
+
+# # Upload files
+resource "google_storage_bucket_object" "parquet_files" {
+  for_each = fileset("${path.module}/assets/parquet/", "*")
+
+  bucket = google_storage_bucket.raw_bucket.name
+  name   = each.value
+  source = "${path.module}/assets/parquet/${each.value}"
+
+}
+
+# # Create a BigQuery external table
+resource "google_bigquery_table" "tbl_edw_taxi" {
+  dataset_id          = google_bigquery_dataset.ds_edw.dataset_id
+  table_id            = "taxi_trips"
+  project             = var.project_id
+  deletion_protection = false
+
+  external_data_configuration {
+    autodetect    = true
+    connection_id = "${var.project_id}.${var.region}.ds_connection"
+    source_format = "PARQUET"
+    source_uris   = ["gs://${google_storage_bucket.raw_bucket.name}/taxi-*.Parquet"]
+
+  }
+
+  depends_on = [
+    google_bigquery_connection.ds_connection,
+    google_storage_bucket.raw_bucket,
+    google_storage_bucket_object.parquet_files
+  ]
+}
+
+# Load Queries for Stored Procedure Execution
+# # Load Lookup Data Tables
+data "template_file" "sp_provision_lookup_tables" {
```

Comment:
nit: you can use https://developer.hashicorp.com/terraform/language/functions/templatefile